code/validation-phase-analysis.R

```diff
@@ -84,5 +84,13 @@ for(forecast_date in validation_forecast_dates){
   
   ##  - fit all models
   
+  model1_solo <- arima_hosp_forecasts(hosp_data)
+
+  model2_reportcasefinal <- arima_hosp_forecasts(hosp_data, report_case_final)
+  
+  model3a_testcasefinal <- arima_hosp_forecasts(hosp_data, test_case_final)
+  
+  model3b_testcaserealtime <- arima_hosp_forecasts(hosp_data, test_case_realtime)
+    
```

Comment:
Is the idea that in the validation phase, we'll fit all variations on models for some possible combinations of `p, d, P, D`? If so, suggest a modification here along these lines:

```suggestion
  # this code should actually go earlier, outside the main for loop; limitation of PR reviews is that I can't do that in this suggestion
  # set up variations on sarima specifications to consider in validation phase
  sarima_variations <- expand.grid(p=0:3, d=0:1, P=0:3, D=0:1)
  for (i in seq_len(nrow(sarima_variations))) {
    p <- sarima_variations$p[i]
    d <- sarima_variations$d[i]
    P <- sarima_variations$P[i]
    D <- sarima_variations$D[i]
    
    model1_solo <- arima_hosp_forecasts(hosp_data, p=p, d=d, P=P, D=D)
    # save model1_solo results

    model2_reportcasefinal <- arima_hosp_forecasts(hosp_data, report_case_final, p=p, d=d, P=P, D=D)
    # save model2_reportcasefinal results
  
    model3a_testcasefinal <- arima_hosp_forecasts(hosp_data, test_case_final, p=p, d=d, P=P, D=D)
    # save model3a_testcasefinal results
  
    model3b_testcaserealtime <- arima_hosp_forecasts(hosp_data, test_case_realtime, p=p, d=d, P=P, D=D)
    # save model3b_testcaserealtime results
  }
```